File: musifanAngular/src/app/models/album.ts
import { Byte } from '@angular/compiler/src/util';
import { Chanson } from './chanson';

export class Album {
  public constructor(
    private _id?: number | undefined,
    private _titre?: string | undefined,
    private _date?: Date | undefined,
    private _photo?: Byte[] | undefined,
    private _chansons?: Chanson[] | undefined,
    private _artistes?: Artiste[] | undefined
  ) {}

  /**
   * Getter id
   * @return {number }
   */
  public get id(): number | undefined {
    return this._id;
  }

  /**
   * Getter titre
   * @return {string }
   */
  public get titre(): string | undefined {
    return this._titre;
  }

  /**
   * Getter date
   * @return {Date }
   */
  public get date(): Date | undefined {
    return this._date;
  }

  /**
   * Getter photo
   * @return {Byte[] }
   */
  public get photo(): Byte[] | undefined {
    return this._photo;
  }

  /**
   * Getter chansons
   * @return {Chanson[] }
   */
  public get chansons(): Chanson[] | undefined {
    return this._chansons;
  }

  /**
   * Getter artistes
   * @return {Artiste[] }
   */
  public get artistes(): Artiste[] | undefined {
    return this._artistes;
  }

  /**
   * Setter id
   * @param {number } value
   */
  public set id(value: number | undefined) {
    this._id = value;
  }

  /**
   * Setter titre
   * @param {string } value
   */
  public set titre(value: string | undefined) {
    this._titre = value;
  }

  /**
   * Setter date
   * @param {Date } value
   */
  public set date(value: Date | undefined) {
    this._date = value;
  }

  /**
   * Setter photo
   * @param {Byte[] } value
   */
  public set photo(value: Byte[] | undefined) {
    this._photo = value;
  }

  /**
   * Setter chansons
   * @param {Chanson[] } value
   */
  public set chansons(value: Chanson[] | undefined) {
    this._chansons = value;
  }

  /**
   * Setter artistes
   * @param {Artiste[] } value
   */
  public set artistes(value: Artiste[] | undefined) {
    this._artistes = value;
  }
}

File: musifan/src/main/java/musifan/musifan/dto/Album.java
package musifan.musifan.dto;

import java.util.HashSet;
import java.util.Set;

import musifan.musifan.entity.LigneAlbum;

public class Album {
	
	private Long id;
	private String titre;
	private Byte photo;
	private Set<Chanson> chansons;
	private Set<Artiste> artistes = new HashSet<Artiste>();
	
	public Album() {
	}

	public Album(musifan.musifan.entity.Album album) {
		this.id = album.getId();
		this.titre = album.getTitre();
		this.photo = album.getPhoto();
		this.chansons = album.getChansons();
		
		for(LigneAlbum la : album.getLignesAlbums()) {
			artistes.add(la.getId().getArtiste());
		}
		
	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getTitre() {
		return titre;
	}

	public void setTitre(String titre) {
		this.titre = titre;
	}

	public Byte getPhoto() {
		return photo;
	}

	public void setPhoto(Byte photo) {
		this.photo = photo;
	}

	public Set<Chanson> getChansons() {
		return chansons;
	}

	public void setChansons(Set<Chanson> chansons) {
		this.chansons = chansons;
	}

	public Set<Artiste> getArtistes() {
		return artistes;
	}

	public void setArtistes(Set<Artiste> artistes) {
		this.artistes = artistes;
	}
	
	
	
	
}
